pyteal/compiler/compiler_test.py

```diff
@@ -2193,3 +2193,457 @@ def access_b4_store(magic_num: pt.abi.Uint64, *, output: pt.abi.Uint64):
+btoi
+store 43
+txna ApplicationArgs 15
+store 44
+load 44
+intc_0 // 0
+extract_uint64
+store 45
+load 44
+pushint 8 // 8
+extract_uint64
+store 46
+load 30
+load 31
+load 32
+load 33
+load 34
+load 35
+load 36
+load 37
+load 38
+load 39
+load 40
+load 41
+load 42
+load 43
+load 45
+load 46
+callsub alllaidtoargs_5
+store 47
+bytec_0 // 0x151f7c75
+load 47
+itob
+concat
+log
+intc_1 // 1
+return
+main_l8:
+txn OnCompletion
+intc_0 // NoOp
+==
+txn ApplicationID
+intc_0 // 0
+!=
+&&
+assert
+txna ApplicationArgs 1
+btoi
+store 24
+txna ApplicationArgs 2
+btoi
+store 25
+load 24
+load 25
+callsub mod_4
+store 26
+bytec_0 // 0x151f7c75
+load 26
+itob
+concat
+log
+intc_1 // 1
+return
+main_l9:
+txn OnCompletion
+intc_0 // NoOp
+==
+txn ApplicationID
+intc_0 // 0
+!=
+&&
+assert
+txna ApplicationArgs 1
+btoi
+store 18
+txna ApplicationArgs 2
+btoi
+store 19
+load 18
+load 19
+callsub div_3
+store 20
+bytec_0 // 0x151f7c75
+load 20
+itob
+concat
+log
+intc_1 // 1
+return
+main_l10:
+txn OnCompletion
+intc_0 // NoOp
+==
+txn ApplicationID
+intc_0 // 0
+!=
+&&
+assert
+txna ApplicationArgs 1
+btoi
+store 12
+txna ApplicationArgs 2
+btoi
+store 13
+load 12
+load 13
+callsub mul_2
+store 14
+bytec_0 // 0x151f7c75
+load 14
+itob
+concat
+log
+intc_1 // 1
+return
+main_l11:
+txn OnCompletion
+intc_0 // NoOp
+==
+txn ApplicationID
+intc_0 // 0
+!=
+&&
+assert
+txna ApplicationArgs 1
+btoi
+store 6
+txna ApplicationArgs 2
+btoi
+store 7
+load 6
+load 7
+callsub sub_1
+store 8
+bytec_0 // 0x151f7c75
+load 8
+itob
+concat
+log
+intc_1 // 1
+return
+main_l12:
+txn OnCompletion
+intc_0 // NoOp
+==
+txn ApplicationID
+intc_0 // 0
+!=
+&&
+assert
+txna ApplicationArgs 1
+btoi
+store 0
+txna ApplicationArgs 2
+btoi
+store 1
+load 0
+load 1
+callsub add_0
+store 2
+bytec_0 // 0x151f7c75
+load 2
+itob
+concat
+log
+intc_1 // 1
+return
+
+// add
+add_0:
+store 4
+store 3
+load 3
+load 4
++
+store 5
+load 5
+retsub
+
+// sub
+sub_1:
+store 10
+store 9
+load 9
+load 10
+-
+store 11
+load 11
+retsub
+
+// mul
+mul_2:
+store 16
+store 15
+load 15
+load 16
+*
+store 17
+load 17
+retsub
+
+// div
+div_3:
+store 22
+store 21
+load 21
+load 22
+/
+store 23
+load 23
+retsub
+
+// mod
+mod_4:
+store 28
+store 27
+load 27
+load 28
+%
+store 29
+load 29
+retsub
+
+// all_laid_to_args
+alllaidtoargs_5:
+store 63
+store 62
+store 61
+store 60
+store 59
+store 58
+store 57
+store 56
+store 55
+store 54
+store 53
+store 52
+store 51
+store 50
+store 49
+store 48
+load 48
+load 49
++
+load 50
++
+load 51
++
+load 52
++
+load 53
++
+load 54
++
+load 55
++
+load 56
++
+load 57
++
+load 58
++
+load 59
++
+load 60
++
+load 61
++
+load 62
++
+load 63
++
+store 64
+load 64
+retsub""".strip()
+    assert expected_ap == actual_ap_compiled
+
+    expected_csp = """#pragma version 6
+intcblock 0
+txn NumAppArgs
+intc_0 // 0
+==
+bnz main_l2
+err
+main_l2:
+txn ApplicationID
+intc_0 // 0
+!=
+assert
+pushint 1 // 1
+return""".strip()
+    assert expected_csp == actual_csp_compiled
```

Comment:
It would be great to see a few more things tested here. Cases I'm interested in seeing:
1. A method that appears in the approval and clear state programs
2. A method that appears only in the clear state program
3. A method that works with app creation
4. A router with no bare app calls
5. A router with bare app calls for clear state + another action

Reply:
I updated the compiler tests to meet these requirements, let me know if we are satisfied with the improvement.

Reply:
got it, my bad for missing the case. adding for case 2 here.